airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -355,17 +361,32 @@ public WebBackendConnectionRead webBackendUpdateConnection(final WebBackendConne
     final List<UUID> operationIds = updateOperations(webBackendConnectionUpdate);
     final ConnectionUpdate connectionUpdate = toConnectionUpdate(webBackendConnectionUpdate, operationIds);
 
-    ConnectionRead connectionRead;
-    final boolean needReset = MoreBooleans.isTruthy(webBackendConnectionUpdate.getWithRefreshedCatalog());
+    final UUID connectionId = webBackendConnectionUpdate.getConnectionId();
+
+    final ConfiguredAirbyteCatalog existingConfiguredCatalog =
+        configRepository.getConfiguredCatalogForConnection(connectionId);
+    final io.airbyte.protocol.models.AirbyteCatalog existingCatalog = CatalogHelpers.configuredCatalogToCatalog(existingConfiguredCatalog);
+    final AirbyteCatalog apiExistingCatalog = CatalogConverter.toApi(existingCatalog);
+    final AirbyteCatalog newAirbyteCatalog = webBackendConnectionUpdate.getSyncCatalog();
+    final CatalogDiff catalogDiff = connectionsHandler.getDiff(apiExistingCatalog, newAirbyteCatalog);
+
+    final List<StreamDescriptor> apiStreamsToReset = getStreamsToReset(catalogDiff);
+    List<io.airbyte.protocol.models.StreamDescriptor> streamsToReset =
+        apiStreamsToReset.stream().map(ProtocolConverters::streamDescriptorToProtocol).toList();
 
+    ConnectionRead connectionRead;
     connectionRead = connectionsHandler.updateConnection(connectionUpdate);
 
-    if (needReset) {
+    if (!streamsToReset.isEmpty()) {
+      final ConnectionIdRequestBody connectionIdRequestBody = new ConnectionIdRequestBody().connectionId(connectionId);
+      final ConnectionStateType stateType = getStateType(connectionIdRequestBody);
+
+      if (stateType == ConnectionStateType.LEGACY || stateType == ConnectionStateType.NOT_SET || stateType == ConnectionStateType.GLOBAL) {
```

Comment:
The plan was to change this to remove the `|| stateType == ConnectionStateType.GLOBAL` once the postgres source is updated with the logic for handling resets of individual streams in the CDC case (I think [this PR](https://github.com/airbytehq/airbyte/pull/14399)), right?

I am concerned this could lead so some issues, such as the following scenario:
- A user updates their platform to have the change to remove `|| stateType == ConnectionStateType.GLOBAL` from this line of code
- This user is still using the version of the postgres source that emits GLOBAL state for CDC, but does not have that individual stream reset change (i.e. the current version of the postgres source 0.4.30)
- The user then performs an action that results in a partial reset of the connection

In this case, since the postgres source doesn't have that new logic, it could lead to bad behavior, e.g. the source just continuing to sync all streams from the current cdc cursor, or not syncing the streams that were reset at all.

@subodh1810 @benmoriceau @gosusnp FYI, I'm not super sure how we should address this issue

Reply:
> This user is still using the version of the postgres source that emits GLOBAL state for CDC, but does not have that individual stream reset change (i.e. the current version of the postgres source 0.4.30)

@lmossman can we force them to migrate to the right postgres connector with the [seed](https://github.com/airbytehq/airbyte/blob/master/airbyte-config/init/src/main/resources/seed/source_definitions.yaml#L750) file? If they don't downgrade it should ensure that we will use the right version. There might still be an issue with companies that are using the public connectors as a custom one. I think that datadog is doing that.

Reply:
Yeah that's a good point, I had thought that we had logic to prevent updating a connector if it was actively being used in a deployment, but it looks like we actually do update connectors even if they are in use https://github.com/airbytehq/airbyte/blob/900c21230efccb2d8ed269c52fc7a3d5f6d37b94/airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java#L1812-L1817

For the custom connector case, we may just have to try to communicate that those will have to be rebuilt on the latest version